### ISSUE TITLE
App is muted and missing from EarTrumpet when previously redirected to an output which is unavailable

### ISSUE THREAD

**Issue by mckress5433** [File-New-Project/EarTrumpet#403]
This issue has happened to me multiple times but I couldn't figure out why. 
Here is what happens:
1) My laptop is connected to a TV
2) I have firefox play audio through the TV using EarTrumpet
3) I close firefox and turn off my laptop, then disconnect from the TV
4) I turn on my laptop and open firefox
5) Audio isn't playing so I try switch firefox back to my laptop speakers
6) Firefox is not showing up in eartrumpet
7) I connect laptop to the same TV and firefox is then present and can be moved to laptop speakers

From my experience, I have to move firefox audio to my laptop speakers if I want to have audio after disconnecting from an audio device. Problem is that I "left" my audio on a TV I no longer have access to and thus I cannot get it back. I've tried uninstalling both EarTrumpet and Firefox completely then reinstalling them but this didn't fix the issue. I've also uninstalled and reinstalled my audio driver but this didn't work either. I haven't lost audio from any other program because of this issue, it seems to only affect firefox. 

Hopefully this is enough information for my issue to be understood. I'll be happy to provide more information and try other approaches to solve this problem.

**Comment by davuxcom** [comment]
Type `Win+X`, select `Settings` and then`System -> Sound -> App volume and device preferences`.

This page lets you set the output for apps or clear the system configuration (`Reset` button at the bottom).  You can use the `Reset` button to fix your issue if the TV output device isn't available.  

We use the same API to change devices, and Windows has some known bugs around this device redirection scenario.  In this specific case I think the expectation is that Firefox would detect that it has a persisted output which is currently unavailable, and clear it (causing the audio to go to the default device).  But due to the way Microsoft rolled out this functionality, apps don't expect to need to do this and thus don't.

From an EarTrumpet perspective, we don't know that Firefox is even playing audio since the audio stack hasn't told us anything about Firefox, so I'm not seeing any good way to mitigate this behavior right now.

**Comment by mckress5433** [comment]
Thank you so much! This fixed my problem!

**Comment by Saint146** [comment]
Hello, thank all contributors for their job!
Actually I have the same problem, and the workaround didn't help me. The app just isn't present even in this windows preferences screen. I can't find it anywhere. I even tried to uninstall EarTrumpet and nothing.

Now I got it that it's a windows bug but maybe somebody has another workaround?
Or maybe EarTrumpet itself can redirect apps in some... different fashion so they don't vanish? :)
(Sorry, didn't look into the code yet.)

**Comment by meghdokania** [comment]
Thanks, it fixed my issue....

**Comment by ririchi2** [comment]
> Thanks, you helped me too !

**Comment by Telekors** [comment]
Thank you! Holy shit you’d think Microsoft would make that easier to find

**Comment by vgullotta8** [comment]
Hello,

I recently installed Eartrumpet on my windows 11 laptop. It was working great until I had to reboot. Now most of my apps won't make any sound, won't show up in Eartrumpet's list, I have been googling all morning trying to get outlook notifications to make noise again so I'm not late to another meeting. I have tried the win x -> settings -> sound -> reset all, it was no help. I have restarted Eartrumpet a ton of times and now it shows no devices at all... This app was great until that first reboot, and now it is disastrous. Please help.


ok update, I found this https://github.com/File-New-Project/EarTrumpet/issues/574 which was helpful. I am windows 11, and I am not allowed to run batch files on work machines, so I made it a PS script, but it needed a bit of adjusting:


taskkill /F /IM eartrumpet* /T
timeout /T 3 /NOBREAK
Start-Process -FilePath "C:\EarTrumpet.lnk"


change c:\Eartrumpet.lnk to the location of your shortcut

**Comment by riverar** [comment]
EarTrumpet wouldn't be the no-audio culprit here as you did a reset of all audio settings in Windows already. Double check your connected devices perhaps?

**Comment by riverar** [comment]
@vgullotta8 EarTrumpet wouldn't be the no-audio culprit here as you did a reset of all audio settings in Windows already. Double check your connected devices perhaps?

The sleep bug, however, is known. Thanks for reporting that.

**Comment by vgullotta8** [comment]
It is definitely eartrumpet causing the issues, this never happened until it was installed and my system rebooted for windows updates. After that it had lost everything sound related. Nothing has been played with sound-wise since I got the laptop until I installed Eartrumpet, I am pretty convinced it is related to the issues I'm having. This powershell script restarts it and seems to help, so I will use it for now, but the app needs an update, this bug is rough, I shouldn't need a powershell restart script for an app to function, it should install as a service with a delayed start from the beginning IMO

**Comment by riverar** [comment]
@vgullotta8 EarTrumpet is mostly a _view-only_ app that displays audio sessions in Windows. Whether it's running or not has no real bearing on audio coming out of your machine. Did you resolve the apps-not-making-sound issue you reported earlier? (If so, try terminating EarTrumpet and note the apps still output audio confirming my point.)

**Comment by vgullotta8** [comment]
I tried changing outlook reminders to a different sound, and it is still silent. Definitely getting frustrated now. Even with eartrumpet  not running, I get no reminder sounds, all other outlook sounds are working though.

**Comment by riverar** [comment]
@vgullotta8 Try doing a Start search for `troubleshoot sound` and launch the "Find and fix problems with playing sound" troubleshooter. That should get you back on your feet. Also wondering if your earlier updates may have included audio drivers that are now having some issues?

**Comment by riverar** [comment]
@vgullotta8 Oh so it's just the outlook reminder sound? Are you using legacy Outlook or that new one?

![Image](https://github.com/user-attachments/assets/2322e4aa-bc10-4db5-9690-02aaa94ef31f)

**Comment by vgullotta8** [comment]
Legacy outlook, I have been through the options so many times. It is checked, I have changed the reminder.wav file to a number of others in c:\windows\media\ but nothing plays when the reminder pops up.

![Image](https://github.com/user-attachments/assets/ba512c4f-50df-4e84-a5b1-63d61d8180b0)

**Comment by riverar** [comment]
@vgullotta8 Maybe check Settings > System > Notifications > Allow notifications to play sounds

![Image](https://github.com/user-attachments/assets/fe8ef756-4bd0-4b28-a38f-f7865cbb801a)

**Comment by vgullotta8** [comment]
yeah, I have checked here as well. Other outlook sounds work, it is ONLY calendar reminders. 

![Image](https://github.com/user-attachments/assets/b1ffd725-0994-4081-b272-144edf98f0b3)

![Image](https://github.com/user-attachments/assets/37eb6cea-d85f-46b0-99dd-90f498738f19)

**Comment by vgullotta8** [comment]
Ok, I found it. There are so many different places for sound settings in windows it is pretty insane.

https://answers.microsoft.com/en-us/outlook_com/forum/all/play-sound-button-on-notifications-settings-grayed/34d4deb7-e42c-4f4d-84b2-6ec301cf5cde

This helped. I found this check box was greyed out in outlook settings. Took switching to NEW outlook to find this setting btw, so I switched to it, found the setting, it was greyed out and I couldn't change it, but then at the top of the window was a windows notification about troubleshooting notification sounds and I clicked the 3rd link it offered and there was a way to enable it there! So crazy, so aggravated with all of this just to get a sound to work again. I am probably uninstalling EarTrumpet until these bugs are worked out, this was absolutely related to it, there have been no other sound changes done to this system ever, there's no way I accidentally clicked something or anything like that. I know users are dumb etc. I am a very experienced senior level admin, this was absolutely an Eartrumpet issue working with outlook. It may be that Outlook is the main culprit simply because it is a trash application, but whatever, it's fixed now lol

**Comment by vgullotta8** [comment]
ok, so I have another issue that maybe you can help with. Ever since resetting sound settings in pretty much every place I can think of trying to fix the notifications issue, I now have an audio notification every time I click on a folder in outlook. Any idea how to get rid of that? It only seems to happen in Old outlook. I sent to windows sounds and thought I would just change that sound to nothing, but it does not seem to exist there. I am so frustrated at this point, I just want to smash everything.

**Comment by vgullotta8** [comment]
aha, never mind I found it. For anyone who needs:

Outlook -> Options -> Accessibility -> uncheck "Provide feedback with sound"


I must have checked this option during my troubleshooting the reminder notification sound

**Comment by vgullotta8** [comment]
welp, it stopped working on new notifications again. I am done. So frustrated I think this is going to be a re-install of windows.....

**Comment by riverar** [comment]
Doing some cleanup and closing as this issue is stale; please create a new issue if you're still having a problem with this. Thanks!
